### ISSUE TITLE
Trying to initialize a 2D (multidimensional) array with a 1D literal quietly results in a 1D array.

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#20676]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Related:

* #20675

The problem described here only happens with _type constraints_ in the context of _variable assignments_.
A direct cast would result in the same error as in #20675

```powershell
([int[,]]  $arr = (1..3)).GetType().Name
```

### Expected behavior

```console
An error message indicating that initializing multidimensional arrays with (jagged) array literals isn't supported.
```

### Actual behavior

```console
Int32[]


That is, a _1D_ (flat) array rather then the requested 2D one was quietly assigned.
```

### Environment data

```powershell
PowerShell 7.4.0-rc.1
```

**Comment by 237dmitry** [comment]
Should multidimensional array be a fixed size?

```powershell
$arr = New-Object 'int32[,]' 3,1
$arr[0,0] = 0
$arr[1,0] = 1
$arr[2,0] = 2
$arr.GetType().Name
```

**Comment by mklement0** [comment]
@237dmitry, yes: _all_ arrays are fixed-size data structures, irrespective of their dimensionality.
(In a _jagged array_, e.g. `$a = (1..2), (3..4)`, replacing one of the elements with a differently sized array is possible, e.g., `$a[0] = 1..3`, but that is an operation that changes an array _element_, not the array itself.)
